### PR TITLE
Update accordion from backend

### LINKED ISSUE #2668: Be able to update accordion label from backend
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Would be nice if developers could update the accordion label from the backend. 

Example use case,

You have a drop down with dataset names on the left hand side and information about the dataset in different accordion components. You'd like the accordion title to reflect the current dataset selected.

For example, in the UI below, it might be nice if instead of "Dataset Description" it read "<dataset-name> description" or "Label Distribution for <dataset-name>"

image

**Describe the solution you'd like**  
```python
gr.Accordion.update(label=<new-label>)
```

**Additional context**

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #2668 .

I don't think I'll be able to address any feedback until I come back so feel free to push commits + merge without me!

Would be good to add a front-end test for this as well.

To test:

```python
import gradio as gr

with gr.Blocks() as demo:
    with gr.Accordion(label="Open for greeting", open=False) as accordion:
        gr.Textbox("Hello!")
    open_btn = gr.Button(value="Open Accordion")
    close_btn = gr.Button(value="Close Accordion")
    open_btn.click(
        lambda: gr.Accordion.update(open=True, label="Open Accordion"),
        inputs=None,
        outputs=[accordion],
    )
    close_btn.click(
        lambda: gr.Accordion.update(open=False, label="Closed Accordion"),
        inputs=None,
        outputs=[accordion],
    )
demo.launch()
```

update_accordion

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @freddyaboulton! Is this ready to be reviewed or still a draft PR?

**Comment by freddyaboulton** [comment]
@abidlabs Opened now! Was waiting for CI

**Comment by pngwn** [comment]
Probably a front end issue. Will take a look.

**Comment by freddyaboulton** [comment]
@abidlabs I think I fixed the inner accordion issue! Note that when you update the inner accordion state when the parent accordion is closed, the inner accordion has the correct state when you open the parent accordion.

update_accordion_inner

@pngwn Does this look good to you?

File: ui/packages/app/src/components/Accordion/Accordion.svelte [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	export let visible: boolean = true;
 	export let open: boolean = true;
 
-	let _open = open;
+	$: _open = open;
```
Comment (pngwn) [inline ui/packages/app/src/components/Accordion/Accordion.svelte:8]: Why did this ever exist in the first place? It is just assigning the value to an intermediary variable? What happens if you just remove the `_open` and use `open` directly?

Comment (pngwn) [reply ui/packages/app/src/components/Accordion/Accordion.svelte:8]: ah.

**Review by pngwn — APPROVED** [review]

**Comment by freddyaboulton** [comment]
Thank you all for the reviews 🤗

**Comment by malin-howto** [comment]
I have multiple Accordions in my code, and I want to ensure that when I toggle one Accordion open, the others automatically close. Here's the current code I have:

`

    with gr.Accordion("1") as acc_1:
        pass

    with gr.Accordion("2") as acc_2:
        pass

    with gr.Accordion("3") as acc_3:
        pass

    with gr.Accordion("4") as acc_4:
        pass
`

**Pull request merged**